### ISSUE TITLE
static analysis of extract

### ISSUE THREAD

**Issue by tillmo** [spechub/Hets#1587]
We need this urgently for the OMG meeitng next week.
@mcodescu, could you just implement a trivial module extraction that always delivers the whole theory of the original OMS/DG node as module? The DG should be extended by a node for the module with an inclusion theorem link into the original node (similarly to intersections). Also, the original node must be flattenable.
I will in parallel take care of a method for module extraction.

**Comment by tillmo** [comment]
Here is an example:
```
logic OWL

ontology o =
  Class: A
  Class: B
extract
  Class: A
end
```

**Comment by mcodescu** [comment]
The example goes through.

**Comment by tillmo** [comment]
@mcodescu, your implementation work with the above example. However, OMS names are missing. See also 57de75c

**Comment by mcodescu** [comment]
I have committed a fix for node names. Note that the ontology that we extract from is anonymous and will get a name that is visible only internally. 

Regarding the second issue, a commit doing that already exists.

**Comment by tillmo** [comment]
good. Now how to call module extraction (works for OWL only). Try in a `ghci`
```
import Network.HTTP
resp <- simpleHTTP (postRequest "http://owlapi.hets.eu/modularity/http%3A%2F%2Fontohub.org%2Ftones%2Fwww.co-ode.org%2Fpizza.owl/hasbase,Mushroom/manowl/false" )
let resp1 = case resp of Right x -> x 
rspBody resp1
```
The resulting string needs to be stored in a file and then processed with `OWL2.ParseOWLAsLibDefn`.

**Comment by tillmo** [comment]
do you mean #1588 with "second issue"? Then could you please mention the commit there?

**Comment by mcodescu** [comment]
No, second issue I mean 57de75c. I am about to push something for #1588.

**Comment by tillmo** [comment]
OK, but 57de75c is a commit, not an issue.

**Comment by mcodescu** [comment]
Yes, I mean: the change that you pointed out by that commit is taken care of, see the second commit of this branch.

**Comment by tillmo** [comment]
Some easier way of extracting modules, using https://github.com/owlcollab/owltools:
```
owltools ontology.owl --extract-module -d class1 ... classn -o output.owl
```
It seems that the classes have to specified with full URL.
With `-m MODULE_TYPE`, different module extraction methods can be selected. See http://owlapi.sourceforge.net/javadoc/uk/ac/manchester/cs/owlapi/modularity/ModuleType.html for the available MODULE_TYPEs.
